app/src/test/java/ch/hikemate/app/model/route/HikeRoutesRepositoryOverpassTest.kt

```diff
@@ -1011,11 +1040,85 @@ class HikeRoutesRepositoryOverpassTest {
       callbackCapture.firstValue.onResponse(mockCall, simpleResponse)
     }
 
+    hikingRouteProviderRepositoryOverpass.getRouteById(
+        simpleRoutes.first().id,
+        { assertEquals(simpleRoutes.first(), it) },
+        { fail("onFailure shouldn't have been called") })
+  }
+
+  @Test
+  fun getRoutesByIds_hasEmptyListOnEmptyResponse() {
+    val mockCall = mock(Call::class.java)
+    `when`(mockClient.newCall(any())).thenReturn(mockCall)
+
+    val callbackCapture = argumentCaptor<okhttp3.Callback>()
+
+    `when`(mockCall.enqueue(callbackCapture.capture())).then {
+      callbackCapture.firstValue.onResponse(mockCall, emptyResponse)
+    }
+
+    hikingRouteProviderRepositoryOverpass.getRoutesByIds(
+        listOf(simpleRoutes.first().id),
+        { assertEquals(emptyList<HikeRoute>(), it) },
+        { fail("onFailure shouldn't have been called") })
+  }
```

Comment:
Nitpick: what if neither `onSuccess` nor `onFailure` is called? You used `val failCalled = false` or equivalent for success in your other tests, is there a reason why you did not here?

Reply:
Oh wait, that's such a great remark, never thought about this! I'm going to change that!